Sources/TuistDependencies/SwiftPackageManager/Utils/PackageInfoMapper.swift

```diff
@@ -291,6 +291,39 @@ public final class PackageInfoMapper: PackageInfoMapping {
         packageToProject: [String: AbsolutePath],
         swiftToolsVersion: TSCUtility.Version?
     ) throws -> ProjectDescription.Project? {
+        // Hardcoded mapping for some well known libraries, until the logic can handle those properly
+        let productTypes = productTypes.merging(
+            // Force dynamic frameworks
+            Dictionary(
+                uniqueKeysWithValues: [
+                    "Checksum", // https://github.com/rnine/Checksum
+                    "RxSwift", // https://github.com/ReactiveX/RxSwift
+                ].map {
+                    ($0, .framework)
+                }
+            ),
+            uniquingKeysWith: { userDefined, _ in userDefined }
+        )
+        let targetSettings = targetSettings.merging(
+            // Force enable testing search paths
+            Dictionary(
+                uniqueKeysWithValues: [
+                    "Nimble", // https://github.com/Quick/Nimble
+                    "Quick", // https://github.com/Quick/Quick
+                    "RxTest", // https://github.com/ReactiveX/RxSwift
+                    "RxTest-Dynamic", // https://github.com/ReactiveX/RxSwift
+                    "SnapshotTesting", // https://github.com/pointfreeco/swift-snapshot-testing
+                    "TempuraTesting", // https://github.com/BendingSpoons/tempura-swift
+                    "TSCTestSupport", // https://github.com/apple/swift-tools-support-core
+                ].map {
+                    ($0, ["ENABLE_TESTING_SEARCH_PATHS": "YES"])
+                }
```

Comment:
Rather than hardcoding, for most packages we should be able to get the desired result by parsing the project file. For example, for RxSwift we have the following:
```
		C88FA5081C25C44800CCFEA4 /* Build configuration list for PBXNativeTarget "RxTest" */ = {
			isa = XCConfigurationList;
			buildConfigurations = (
				C88FA5091C25C44800CCFEA4 /* Debug */,
				C88FA50A1C25C44800CCFEA4 /* Release */,
				C88FA50B1C25C44800CCFEA4 /* Release-Tests */,
			);
			defaultConfigurationIsVisible = 0;
			defaultConfigurationName = Release;
		};
```
```
		C88FA5091C25C44800CCFEA4 /* Debug */ = {
			isa = XCBuildConfiguration;
			buildSettings = {
				<snip>
				ENABLE_TESTING_SEARCH_PATHS = YES;
				<snip>
			};
			name = Debug;
		};
```

From this we can infer than the "RxTest" target should have the setting enabled.

Reply:
I think our logic should be based on the Package.swift content, as the Xcode project might not be present or even be something completely different from the package

Reply:
Could you point em at the `Package.swift` one? 🙏

Reply:
Sorry, meant Project.swift.

Reply:
I don't think we should use neither Xcode project nor Project.swift, as this is a mapper for SPM dependencies, hence it should work on the `Package.swift`